ios/Classes/SwiftFpjsProPlugin.swift

```diff
@@ -14,38 +14,57 @@ public class SwiftFpjsProPlugin: NSObject, FlutterPlugin {
     public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
         if (call.method == "init") {
             if let args = call.arguments as? Dictionary<String, Any>,
-               let token = args["apiToken"] as? String {
-                let region = args["region"] as? String
+                let token = args["apiToken"] as? String {
+                let region = parseRegion(passedRegion: args["region"] as? String, endpoint: args["endpoint"] as? String)
 
-                var endpoint: URL?
-                if let endpointString = args["endpoint"] as? String {
-                    endpoint = URL(string: endpointString)
-                }
-
-                initFpjs(token: token, endpoint: endpoint, region: region)
+                initFpjs(token: token, region: region)
                 result("Successfully initialized FingerprintJS Pro Client")
             } else {
                 result(FlutterError.init(code: "errorApiToken", message: "missing API Token", details: nil))
             }
         } else if (call.method == "getVisitorId") {
             let args = call.arguments as? Dictionary<String, Any>
-            let tags = args?["tags"] as? FingerprintJSProClient.Tags
+            let tags = args?["tags"] as? [String: JSONTypeConvertible]
+            var metadata = Metadata()
+            
+            tags?.forEach({ (tagName: String, tagValue: JSONTypeConvertible) in
+                metadata.setTag(tagValue, forKey: tagName)
+            })
+
+            getVisitorId(metadata, result)
+        }
+    }
 
-            getVisitorId(tags: tags, { (visitorId: String) -> Void in result(visitorId) })
+    private func parseRegion(passedRegion: String?, endpoint: String?) -> Region {
+        var region: Region
+        switch passedRegion {
+        case "eu":
+            region = .eu
+        case "ap":
+            region = .ap
+        default:
+            region = .global
         }
+
+        if let endpointString = endpoint {
+            region = .custom(domain: endpointString)
+        }
+
+        return region
     }
 
-    private func initFpjs(token: String, endpoint: URL? = nil, region: String? = nil) {
-        fpjsClient = FingerprintJSProFactory.getInstance(token: token, endpoint: endpoint, region: region)
+    private func initFpjs(token: String, region: Region) {
+        let configuration = Configuration(apiKey: token, region: region)
+        fpjsClient = FingerprintProFactory.getInstance(configuration)
     }
 
-    private func getVisitorId(tags: FingerprintJSProClient.Tags?, _ handler: @escaping (String) -> Void) {
-        fpjsClient?.getVisitorId { result in
-            switch result {
-            case let .failure(error):
-                handler(error.localizedDescription)
-            case let .success(visitorId):
-                handler(visitorId)
+    private func getVisitorId(_ metadata: Metadata?, _ result: @escaping FlutterResult) {
```

Comment:
The method never calls the closure when `fpjsClient` is `nil`. It was the case before but we should probably do something about it (`guard let client = fpjsClient else { result(<failure>) }` maybe?).